src/asm/symbol.cpp

```diff
@@ -53,6 +53,12 @@ bool sym_IsPC(Symbol const *sym) {
 	return sym == PCSymbol;
 }
 
+bool sym_IsDotScope(std::string const &symName) {
+	// Label scopes `.` and `..` are the only nonlocal identifiers that start with a dot.
+	// Three or more dots are considered a nonsensical local label.
+	return symName == "." || symName == "..";
```

Comment:
If/when we *do* implement arbitrarily nested label scopes, then this can just become `return identifier.find_first_not_of('.') == identifier.npos;` and it should Just Work. (That shouldn't reintroduce the `DEF .... EQU 42` bug because this PR also added a `!sym_IsDotScope(identifier)` check for special handling of colon `identifier:`.)